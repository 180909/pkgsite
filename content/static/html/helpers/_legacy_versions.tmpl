<!--
  Copyright 2020 The Go Authors. All rights reserved.
  Use of this source code is governed by a BSD-style
  license that can be found in the LICENSE file.
-->

{{/* . is internal/frontend.VersionsDetails */}}

{{define "legacy_versions"}}
  <div class="Versions">
    <table>
      <tr><th colspan="3"><h2>Versions in this module</h2></th></tr>
      {{template "module_list" .ThisModule}}
      {{if .IncompatibleModules}}
        <tr><th colspan="3"><h2>Incompatible versions in this module</h2></th></tr>
        {{template "module_list_incompatible" .IncompatibleModules}}
      {{end}}
    </table>

    {{if .OtherModules}}
      <h2>Other modules containing this package</h2>
      {{range .OtherModules}}
        <div><a href="/{{.}}">{{.}}</a></div>
      {{end}}
    {{end}}
  </div>
{{end}}


{{/* . is []*internal/frontend.VersionList */}}

{{define "module_list_incompatible"}}
  {{range $major := .}}
     {{range $i, $v := $major.Versions}}
       <tr>
         <td>
           {{if eq $i 0 }}
             <div class="Versions-major">
               {{if $major.Deprecated}}(Deprecated{{with $major.DeprecationComment}}: {{.}}{{end}}){{end}}
             </div>
           {{end}}
         </td>
         <td>
           <a href="{{$v.Link}}">{{$v.Version}}</a>
           {{if $v.Retracted}}(Retracted{{with .RetractionRationale}}: {{.}}){{end}}{{end}}
         </td>
         <td>
           <div class="Versions-commitTime">{{$v.CommitTime}}</div>
         </td>
       </tr>
     {{end}}
  {{end}}
{{end}}

{{define "module_list"}}
  {{range $major := .}}
     {{range $i, $v := $major.Versions}}
       <tr>
         <td>
           {{if eq $i 0 }}
             <div class="Versions-major">
               {{$major.Major}}
               {{if $major.Deprecated}}(Deprecated{{with $major.DeprecationComment}}: {{.}}{{end}}){{end}}
             </div>
           {{end}}
         </td>
         <td>
           <a href="{{$v.Link}}">{{$v.Version}}</a>
           {{if $v.Retracted}}(Retracted{{with .RetractionRationale}}: {{.}}){{end}}{{end}}
         </td>
         <td>
           <div class="Versions-commitTime">{{$v.CommitTime}}</div>
           {{if $v.Symbols }}
             <div class="Versions-symbols">
             <div class="Versions-symbolsHeader">Changes in this version</div>
             {{range $v.Symbols}}
               <div class="Versions-symbolSection">
               {{range .}}
                 {{if eq .Kind "Type"}}
                   <div class="Versions-symbolType">
                   {{template "legacy_symbol" .}}
                   {{range .Children}}
                     <div class="Versions-symbolChild">{{template "legacy_symbol" .}}</div>
                   {{end}}
                   </div>
                 {{else}}
                   <div>{{template "legacy_symbol" .}}</div>
                 {{end}}
               {{end}}
               </div>
             {{end}}
           </div>
           {{end}}
         </td>
       </tr>
     {{end}}
  {{end}}
{{end}}

{{define "legacy_symbol"}}
  <div>
     {{if .New}}
       <span class="Versions-symbolBulletNew">+</span><a class="Versions-symbolSynopsis" href="{{.Link}}">{{.Synopsis}}</a>
     {{else}}
       <span class="Versions-symbolOld Versions-symbolSynopsis">{{.Synopsis}}</span>
     {{end}}
     {{if .Builds}}
       <span class="Versions-symbolBuildsDash">â€”</span>
       <span class="Versions-symbolBuilds">{{range $i, $b := .Builds}}{{if $i}}, {{end}}{{$b}}{{end}}</span>
     {{end}}
  </div>
{{end}}
