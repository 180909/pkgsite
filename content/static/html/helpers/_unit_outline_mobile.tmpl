<!--
  Copyright 2021 The Go Authors. All rights reserved.
  Use of this source code is governed by a BSD-style
  license that can be found in the LICENSE file.
-->

{{define "unit_outline_mobile"}}
  <nav class="DocNavMobile js-mobileNav">
    <label for="DocNavMobile-select" class="DocNavMobile-label">
      <svg class="DocNavMobile-selectIcon" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="black" width="18px" height="18px">
        <path d="M0 0h24v24H0z" fill="none"/><path d="M3 9h14V7H3v2zm0 4h14v-2H3v2zm0 4h14v-2H3v2zm16 0h2v-2h-2v2zm0-10v2h2V7h-2zm0 6h2v-2h-2v2z"/>
      </svg>
      <span class="DocNavMobile-selectText js-mobileNavSelectText">Outline</span>
    </label>
    <select id="DocNavMobile-select" class="DocNavMobile-select">
      <option value="">Outline</option>
      <optgroup label="README">
        {{range .ReadmeOutline}}<option value="{{.ID}}">{{.Text}}</option>{{end}}
      </optgroup>
      {{.MobileOutline}}
      <option class="js-sourcefilesOption" value="section-sourcefiles">Source Files</option>
      <option class="js-directoriesOption" value="section-directories">Directories</option>
    </select>
  </nav>
{{end}}
