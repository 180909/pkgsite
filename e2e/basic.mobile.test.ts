/**
 * @license
 * Copyright 2020 The Go Authors. All rights reserved.
 * Use of this source code is governed by a BSD-style
 * license that can be found in the LICENSE file.
 */

import { Page } from 'puppeteer';

import './global-types';
import { testcases } from './basic.testcases';
import * as pg from './helpers/page';

let page: Page;

beforeAll(async () => {
  page = await pg.newPage();
  await page.setViewport({ width: 411, height: 731 });
});

afterAll(async () => {
  await page.close();
});

testcases('$name accessibility tree', async ({ path, prepare }) => {
  await page.goto(path);
  if (prepare) await prepare(page);
  const a11yTree = await page.accessibility.snapshot();
  expect(a11yTree).toMatchSnapshot();
});

testcases('$name screenshot', async ({ path, prepare }) => {
  await page.goto(path);
  if (prepare) await prepare(page);
  const image = await page.screenshot({ fullPage: true });
  expect(image).toMatchImageSnapshot();
});

test('no page errors', () => {
  expect(pageErrors).toHaveLength(0);
});
