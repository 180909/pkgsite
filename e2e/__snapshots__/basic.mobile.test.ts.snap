// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`404 with fetch button accessibility tree 1`] = `
Object {
  "children": Array [
    Object {
      "name": "Link to Go Homepage",
      "role": "link",
    },
    Object {
      "name": "Search for a package",
      "role": "button",
    },
    Object {
      "description": "Search for a package",
      "name": "Search for a package",
      "role": "textbox",
    },
    Object {
      "name": "Open navigation.",
      "role": "button",
    },
    Object {
      "name": "The Go Gopher",
      "role": "img",
    },
    Object {
      "level": 3,
      "name": "Oops! We couldn't find “github.com/package/does/not/exist”.",
      "role": "heading",
    },
    Object {
      "name": "Check that you entered the URL correctly, try fetching it following the",
      "role": "text",
    },
    Object {
      "name": "instructions here",
      "role": "link",
    },
    Object {
      "name": ", or request to add “github.com/package/does/not/exist” to pkg.go.dev.",
      "role": "text",
    },
    Object {
      "name": "Request “github.com/package/does/not/exist”",
      "role": "button",
    },
    Object {
      "description": "Why Go",
      "name": "Why Go",
      "role": "link",
    },
    Object {
      "description": "Use Cases",
      "name": "Use Cases",
      "role": "link",
    },
    Object {
      "description": "Case Studies",
      "name": "Case Studies",
      "role": "link",
    },
    Object {
      "description": "Getting Started",
      "name": "Getting Started",
      "role": "link",
    },
    Object {
      "name": "Playground",
      "role": "link",
    },
    Object {
      "name": "Tour",
      "role": "link",
    },
    Object {
      "name": "Stack Overflow",
      "role": "link",
    },
    Object {
      "description": "Discover Packages",
      "name": "Discover Packages",
      "role": "link",
    },
    Object {
      "description": "About",
      "name": "About",
      "role": "link",
    },
    Object {
      "name": "Download",
      "role": "link",
    },
    Object {
      "name": "Blog",
      "role": "link",
    },
    Object {
      "name": "Release Notes",
      "role": "link",
    },
    Object {
      "name": "Brand Guidelines",
      "role": "link",
    },
    Object {
      "name": "Code of Conduct",
      "role": "link",
    },
    Object {
      "description": "Connect",
      "name": "Connect",
      "role": "link",
    },
    Object {
      "name": "Twitter",
      "role": "link",
    },
    Object {
      "name": "GitHub",
      "role": "link",
    },
    Object {
      "name": "Slack",
      "role": "link",
    },
    Object {
      "name": "Meetup",
      "role": "link",
    },
    Object {
      "name": "Gopher in flight goggles",
      "role": "img",
    },
    Object {
      "name": "Copyright",
      "role": "link",
    },
    Object {
      "name": "Terms of Service",
      "role": "link",
    },
    Object {
      "name": "Privacy Policy",
      "role": "link",
    },
    Object {
      "name": "Report an Issue",
      "role": "link",
    },
    Object {
      "name": "golang.org",
      "role": "link",
    },
    Object {
      "name": "Google logo",
      "role": "link",
    },
  ],
  "name": "404 Not Found · pkg.go.dev",
  "role": "WebArea",
}
`;

exports[`badge accessibility tree 1`] = `
Object {
  "children": Array [
    Object {
      "name": "Link to Go Homepage",
      "role": "link",
    },
    Object {
      "name": "Search for a package",
      "role": "button",
    },
    Object {
      "description": "Search for a package",
      "name": "Search for a package",
      "role": "textbox",
    },
    Object {
      "name": "Open navigation.",
      "role": "button",
    },
    Object {
      "level": 1,
      "name": "Create a badge",
      "role": "heading",
    },
    Object {
      "name": "Create a badge to link to pkg.go.dev from your project website or README file.",
      "role": "text",
    },
    Object {
      "name": "Badge",
      "role": "text",
    },
    Object {
      "name": "Go Reference",
      "role": "link",
    },
    Object {
      "name": "URL",
      "role": "text",
    },
    Object {
      "name": "URL",
      "role": "textbox",
    },
    Object {
      "name": "",
      "role": "button",
    },
    Object {
      "name": "The Go Gopher",
      "role": "img",
    },
    Object {
      "name": "Type a pkg.go.dev URL above to create a badge link.",
      "role": "text",
    },
    Object {
      "description": "Why Go",
      "name": "Why Go",
      "role": "link",
    },
    Object {
      "description": "Use Cases",
      "name": "Use Cases",
      "role": "link",
    },
    Object {
      "description": "Case Studies",
      "name": "Case Studies",
      "role": "link",
    },
    Object {
      "description": "Getting Started",
      "name": "Getting Started",
      "role": "link",
    },
    Object {
      "name": "Playground",
      "role": "link",
    },
    Object {
      "name": "Tour",
      "role": "link",
    },
    Object {
      "name": "Stack Overflow",
      "role": "link",
    },
    Object {
      "description": "Discover Packages",
      "name": "Discover Packages",
      "role": "link",
    },
    Object {
      "description": "About",
      "name": "About",
      "role": "link",
    },
    Object {
      "name": "Download",
      "role": "link",
    },
    Object {
      "name": "Blog",
      "role": "link",
    },
    Object {
      "name": "Release Notes",
      "role": "link",
    },
    Object {
      "name": "Brand Guidelines",
      "role": "link",
    },
    Object {
      "name": "Code of Conduct",
      "role": "link",
    },
    Object {
      "description": "Connect",
      "name": "Connect",
      "role": "link",
    },
    Object {
      "name": "Twitter",
      "role": "link",
    },
    Object {
      "name": "GitHub",
      "role": "link",
    },
    Object {
      "name": "Slack",
      "role": "link",
    },
    Object {
      "name": "Meetup",
      "role": "link",
    },
    Object {
      "name": "Gopher in flight goggles",
      "role": "img",
    },
    Object {
      "name": "Copyright",
      "role": "link",
    },
    Object {
      "name": "Terms of Service",
      "role": "link",
    },
    Object {
      "name": "Privacy Policy",
      "role": "link",
    },
    Object {
      "name": "Report an Issue",
      "role": "link",
    },
    Object {
      "name": "golang.org",
      "role": "link",
    },
    Object {
      "name": "Google logo",
      "role": "link",
    },
  ],
  "name": "Badge generation tool · pkg.go.dev",
  "role": "WebArea",
}
`;

exports[`error accessibility tree 1`] = `
Object {
  "children": Array [
    Object {
      "name": "Link to Go Homepage",
      "role": "link",
    },
    Object {
      "name": "Search for a package",
      "role": "button",
    },
    Object {
      "description": "Search for a package",
      "name": "Search for a package",
      "role": "textbox",
    },
    Object {
      "name": "Open navigation.",
      "role": "button",
    },
    Object {
      "name": "The Go Gopher",
      "role": "img",
    },
    Object {
      "level": 3,
      "name": "v1.0-badversion is not a valid semantic version.",
      "role": "heading",
    },
    Object {
      "name": "To search for packages like bad.package,",
      "role": "text",
    },
    Object {
      "name": "click here",
      "role": "link",
    },
    Object {
      "name": ".",
      "role": "text",
    },
    Object {
      "description": "Why Go",
      "name": "Why Go",
      "role": "link",
    },
    Object {
      "description": "Use Cases",
      "name": "Use Cases",
      "role": "link",
    },
    Object {
      "description": "Case Studies",
      "name": "Case Studies",
      "role": "link",
    },
    Object {
      "description": "Getting Started",
      "name": "Getting Started",
      "role": "link",
    },
    Object {
      "name": "Playground",
      "role": "link",
    },
    Object {
      "name": "Tour",
      "role": "link",
    },
    Object {
      "name": "Stack Overflow",
      "role": "link",
    },
    Object {
      "description": "Discover Packages",
      "name": "Discover Packages",
      "role": "link",
    },
    Object {
      "description": "About",
      "name": "About",
      "role": "link",
    },
    Object {
      "name": "Download",
      "role": "link",
    },
    Object {
      "name": "Blog",
      "role": "link",
    },
    Object {
      "name": "Release Notes",
      "role": "link",
    },
    Object {
      "name": "Brand Guidelines",
      "role": "link",
    },
    Object {
      "name": "Code of Conduct",
      "role": "link",
    },
    Object {
      "description": "Connect",
      "name": "Connect",
      "role": "link",
    },
    Object {
      "name": "Twitter",
      "role": "link",
    },
    Object {
      "name": "GitHub",
      "role": "link",
    },
    Object {
      "name": "Slack",
      "role": "link",
    },
    Object {
      "name": "Meetup",
      "role": "link",
    },
    Object {
      "name": "Gopher in flight goggles",
      "role": "img",
    },
    Object {
      "name": "Copyright",
      "role": "link",
    },
    Object {
      "name": "Terms of Service",
      "role": "link",
    },
    Object {
      "name": "Privacy Policy",
      "role": "link",
    },
    Object {
      "name": "Report an Issue",
      "role": "link",
    },
    Object {
      "name": "golang.org",
      "role": "link",
    },
    Object {
      "name": "Google logo",
      "role": "link",
    },
  ],
  "name": "400 Bad Request · pkg.go.dev",
  "role": "WebArea",
}
`;

exports[`home accessibility tree 1`] = `
Object {
  "children": Array [
    Object {
      "name": "Link to Go Homepage",
      "role": "link",
    },
    Object {
      "name": "Open navigation.",
      "role": "button",
    },
    Object {
      "name": "Cartoon gopher typing",
      "role": "img",
    },
    Object {
      "description": "Search for Go packages",
      "focused": true,
      "name": "Search for Go packages",
      "role": "textbox",
    },
    Object {
      "name": "Search",
      "role": "button",
    },
    Object {
      "name": "EXAMPLE SEARCHES:",
      "role": "text",
    },
    Object {
      "name": "“http”",
      "role": "link",
    },
    Object {
      "name": "“command”",
      "role": "link",
    },
    Object {
      "name": "“yaml OR json OR xml”",
      "role": "link",
    },
    Object {
      "name": "Search help",
      "role": "link",
    },
    Object {
      "name": "Frequently asked questions:",
      "role": "text",
    },
    Object {
      "name": "How can I add a package?",
      "role": "link",
    },
    Object {
      "name": "How can I remove a package?",
      "role": "link",
    },
    Object {
      "name": "How can I add a go badge in my README file?",
      "role": "link",
    },
    Object {
      "description": "Why Go",
      "name": "Why Go",
      "role": "link",
    },
    Object {
      "description": "Use Cases",
      "name": "Use Cases",
      "role": "link",
    },
    Object {
      "description": "Case Studies",
      "name": "Case Studies",
      "role": "link",
    },
    Object {
      "description": "Getting Started",
      "name": "Getting Started",
      "role": "link",
    },
    Object {
      "name": "Playground",
      "role": "link",
    },
    Object {
      "name": "Tour",
      "role": "link",
    },
    Object {
      "name": "Stack Overflow",
      "role": "link",
    },
    Object {
      "description": "Discover Packages",
      "name": "Discover Packages",
      "role": "link",
    },
    Object {
      "description": "About",
      "name": "About",
      "role": "link",
    },
    Object {
      "name": "Download",
      "role": "link",
    },
    Object {
      "name": "Blog",
      "role": "link",
    },
    Object {
      "name": "Release Notes",
      "role": "link",
    },
    Object {
      "name": "Brand Guidelines",
      "role": "link",
    },
    Object {
      "name": "Code of Conduct",
      "role": "link",
    },
    Object {
      "description": "Connect",
      "name": "Connect",
      "role": "link",
    },
    Object {
      "name": "Twitter",
      "role": "link",
    },
    Object {
      "name": "GitHub",
      "role": "link",
    },
    Object {
      "name": "Slack",
      "role": "link",
    },
    Object {
      "name": "Meetup",
      "role": "link",
    },
    Object {
      "name": "Gopher in flight goggles",
      "role": "img",
    },
    Object {
      "name": "Copyright",
      "role": "link",
    },
    Object {
      "name": "Terms of Service",
      "role": "link",
    },
    Object {
      "name": "Privacy Policy",
      "role": "link",
    },
    Object {
      "name": "Report an Issue",
      "role": "link",
    },
    Object {
      "name": "golang.org",
      "role": "link",
    },
    Object {
      "name": "Google logo",
      "role": "link",
    },
  ],
  "name": "Home · pkg.go.dev",
  "role": "WebArea",
}
`;

exports[`license policy accessibility tree 1`] = `
Object {
  "children": Array [
    Object {
      "name": "Link to Go Homepage",
      "role": "link",
    },
    Object {
      "name": "Search for a package",
      "role": "button",
    },
    Object {
      "description": "Search for a package",
      "name": "Search for a package",
      "role": "textbox",
    },
    Object {
      "name": "Open navigation.",
      "role": "button",
    },
    Object {
      "level": 1,
      "name": "License Disclaimer",
      "role": "heading",
    },
    Object {
      "name": "The Go website displays license information in order to help users evaluate packages for their intended use. Licenses are detected using heuristics based on their file name and contents. We hope this information is helpful, but this is not legal advice and we do not make any guarantees regarding the accuracy of our license detection.",
      "role": "text",
    },
    Object {
      "name": "If we are not able to detect one of the licenses below, only limited package and module information will be made available. If you are a package author seeking to make your content available on the Go website, please be aware that our detection algorithms can be affected by any modifications of the license text, or by the use of an uncommon license file name.",
      "role": "text",
    },
    Object {
      "name": "We currently use",
      "role": "text",
    },
    Object {
      "name": "github.com/google/licensecheck",
      "role": "link",
    },
    Object {
      "name": "for license detection, and look for licenses in files with the following names: COPYING, COPYING.md, COPYING.markdown, COPYING.txt, LICENCE, LICENCE.md, LICENCE.markdown, LICENCE.txt, LICENSE, LICENSE.md, LICENSE.markdown, LICENSE.txt, LICENSE-2.0.txt, LICENCE-2.0.txt, LICENSE-APACHE, LICENCE-APACHE, LICENSE-APACHE-2.0.txt, LICENCE-APACHE-2.0.txt, LICENSE-MIT, LICENCE-MIT, LICENSE.MIT, LICENCE.MIT, LICENSE.code, LICENCE.code, LICENSE.docs, LICENCE.docs, LICENSE.rst, LICENCE.rst, MIT-LICENSE, MIT-LICENCE, MIT-LICENSE.md, MIT-LICENCE.md, MIT-LICENSE.markdown, MIT-LICENCE.markdown, MIT-LICENSE.txt, MIT-LICENCE.txt, MIT_LICENSE, MIT_LICENCE, UNLICENSE, UNLICENCE. The match is case-insensitive.",
      "role": "text",
    },
    Object {
      "name": "We currently detect and recognize the following licenses:",
      "role": "text",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "0BSD",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "AFL-3.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "AGPL-3.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "AGPL-3.0-only",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "AGPL-3.0-or-later",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "Apache-1.1",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "Apache-2.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "Artistic-2.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "BSD-1-Clause",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "BSD-2-Clause",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "BSD-2-Clause-Patent",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "BSD-2-Clause-Views",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause-Clear",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "BSD-3-Clause-Open-MPI",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "BSD-4-Clause",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "BSD-4-Clause-UC",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "BSL-1.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "BlueOak-1.0.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "CC-BY-3.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "CC-BY-4.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "CC-BY-SA-3.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "CC-BY-SA-4.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "CC0-1.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "EPL-1.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "EPL-2.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "EUPL-1.2",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "GPL-2.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "GPL-2.0-only",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "GPL-2.0-or-later",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "GPL-3.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "GPL-3.0-only",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "GPL-3.0-or-later",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "HPND",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "ISC",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "JSON",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "LGPL-2.1",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "LGPL-2.1-or-later",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "LGPL-3.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "LGPL-3.0-or-later",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "MIT",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "MIT-0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "MPL-2.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "MPL-2.0-no-copyleft-exception",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "NCSA",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "NIST-PD",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "NIST-PD-fallback",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "OSL-3.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "OpenSSL",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "PostgreSQL",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "Python-2.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "UPL-1.0",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "Unlicense",
      "role": "link",
    },
    Object {
      "name": "•",
      "role": "text",
    },
    Object {
      "name": "Zlib",
      "role": "link",
    },
    Object {
      "name": "If you use a package whose license is not detected, please inform the package author. If you are a package author who believes a license for one of your packages should have been detected and was not, please check for discrepancies between your license and the official text. If you still believe there is an error, please",
      "role": "text",
    },
    Object {
      "name": "file an issue",
      "role": "link",
    },
    Object {
      "name": ".",
      "role": "text",
    },
    Object {
      "description": "Why Go",
      "name": "Why Go",
      "role": "link",
    },
    Object {
      "description": "Use Cases",
      "name": "Use Cases",
      "role": "link",
    },
    Object {
      "description": "Case Studies",
      "name": "Case Studies",
      "role": "link",
    },
    Object {
      "description": "Getting Started",
      "name": "Getting Started",
      "role": "link",
    },
    Object {
      "name": "Playground",
      "role": "link",
    },
    Object {
      "name": "Tour",
      "role": "link",
    },
    Object {
      "name": "Stack Overflow",
      "role": "link",
    },
    Object {
      "description": "Discover Packages",
      "name": "Discover Packages",
      "role": "link",
    },
    Object {
      "description": "About",
      "name": "About",
      "role": "link",
    },
    Object {
      "name": "Download",
      "role": "link",
    },
    Object {
      "name": "Blog",
      "role": "link",
    },
    Object {
      "name": "Release Notes",
      "role": "link",
    },
    Object {
      "name": "Brand Guidelines",
      "role": "link",
    },
    Object {
      "name": "Code of Conduct",
      "role": "link",
    },
    Object {
      "description": "Connect",
      "name": "Connect",
      "role": "link",
    },
    Object {
      "name": "Twitter",
      "role": "link",
    },
    Object {
      "name": "GitHub",
      "role": "link",
    },
    Object {
      "name": "Slack",
      "role": "link",
    },
    Object {
      "name": "Meetup",
      "role": "link",
    },
    Object {
      "name": "Gopher in flight goggles",
      "role": "img",
    },
    Object {
      "name": "Copyright",
      "role": "link",
    },
    Object {
      "name": "Terms of Service",
      "role": "link",
    },
    Object {
      "name": "Privacy Policy",
      "role": "link",
    },
    Object {
      "name": "Report an Issue",
      "role": "link",
    },
    Object {
      "name": "golang.org",
      "role": "link",
    },
    Object {
      "name": "Google logo",
      "role": "link",
    },
  ],
  "name": "Licenses · pkg.go.dev",
  "role": "WebArea",
}
`;

exports[`search help accessibility tree 1`] = `
Object {
  "children": Array [
    Object {
      "name": "Link to Go Homepage",
      "role": "link",
    },
    Object {
      "name": "Search for a package",
      "role": "button",
    },
    Object {
      "description": "Search for a package",
      "name": "Search for a package",
      "role": "textbox",
    },
    Object {
      "name": "Open navigation.",
      "role": "button",
    },
    Object {
      "level": 1,
      "name": "Search help",
      "role": "heading",
    },
    Object {
      "name": "You can use symbols or words in your search to make your search results more precise.",
      "role": "text",
    },
    Object {
      "level": 2,
      "name": "Search for an exact match",
      "role": "heading",
    },
    Object {
      "name": "Put a word or phrase inside quotes. For example,",
      "role": "text",
    },
    Object {
      "name": "\\"go cloud\\"",
      "role": "link",
    },
    Object {
      "name": ".",
      "role": "text",
    },
    Object {
      "level": 2,
      "name": "Combine searches",
      "role": "heading",
    },
    Object {
      "name": "Put OR between each search query. For example,",
      "role": "text",
    },
    Object {
      "name": "yaml OR json",
      "role": "link",
    },
    Object {
      "name": ".",
      "role": "text",
    },
    Object {
      "level": 2,
      "name": "Search by package path",
      "role": "heading",
    },
    Object {
      "name": "You can search for a package by its full or partial import path. For example,",
      "role": "text",
    },
    Object {
      "name": "go/packages",
      "role": "link",
    },
    Object {
      "name": ".",
      "role": "text",
    },
    Object {
      "name": "If the query matches a package import path, you will be redirected to the package details page for the latest version of that package. For example,",
      "role": "text",
    },
    Object {
      "name": "golang.org/x/tools/go/packages",
      "role": "link",
    },
    Object {
      "name": ".",
      "role": "text",
    },
    Object {
      "description": "Why Go",
      "name": "Why Go",
      "role": "link",
    },
    Object {
      "description": "Use Cases",
      "name": "Use Cases",
      "role": "link",
    },
    Object {
      "description": "Case Studies",
      "name": "Case Studies",
      "role": "link",
    },
    Object {
      "description": "Getting Started",
      "name": "Getting Started",
      "role": "link",
    },
    Object {
      "name": "Playground",
      "role": "link",
    },
    Object {
      "name": "Tour",
      "role": "link",
    },
    Object {
      "name": "Stack Overflow",
      "role": "link",
    },
    Object {
      "description": "Discover Packages",
      "name": "Discover Packages",
      "role": "link",
    },
    Object {
      "description": "About",
      "name": "About",
      "role": "link",
    },
    Object {
      "name": "Download",
      "role": "link",
    },
    Object {
      "name": "Blog",
      "role": "link",
    },
    Object {
      "name": "Release Notes",
      "role": "link",
    },
    Object {
      "name": "Brand Guidelines",
      "role": "link",
    },
    Object {
      "name": "Code of Conduct",
      "role": "link",
    },
    Object {
      "description": "Connect",
      "name": "Connect",
      "role": "link",
    },
    Object {
      "name": "Twitter",
      "role": "link",
    },
    Object {
      "name": "GitHub",
      "role": "link",
    },
    Object {
      "name": "Slack",
      "role": "link",
    },
    Object {
      "name": "Meetup",
      "role": "link",
    },
    Object {
      "name": "Gopher in flight goggles",
      "role": "img",
    },
    Object {
      "name": "Copyright",
      "role": "link",
    },
    Object {
      "name": "Terms of Service",
      "role": "link",
    },
    Object {
      "name": "Privacy Policy",
      "role": "link",
    },
    Object {
      "name": "Report an Issue",
      "role": "link",
    },
    Object {
      "name": "golang.org",
      "role": "link",
    },
    Object {
      "name": "Google logo",
      "role": "link",
    },
  ],
  "name": "Search Help · pkg.go.dev",
  "role": "WebArea",
}
`;
